### PR TITLE
Default api key header for the ParaSwap client

### PR DESCRIPTION
**Pull request by squadgazzz**
Uses the configured api key by default in the ParaSwap client

### CODE REVIEW

**Review by Mateo-mro — APPROVED** [review]

File: src/infra/dex/paraswap/mod.rs [hunk 1]
```diff
@@ -43,11 +43,21 @@ pub struct Config {
...
 impl ParaSwap {
-    pub fn new(config: Config) -> Self {
-        Self {
-            client: super::Client::new(Default::default(), config.block_stream.clone()),
+    pub fn new(config: Config) -> anyhow::Result<Self> {
+        let mut key = reqwest::header::HeaderValue::from_str(&config.api_key)?;
+        key.set_sensitive(true);
+
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert("x-api-key", key);
```
Comment (fleupold) [inline src/infra/dex/paraswap/mod.rs:51]: before it was upper caps, does this make a difference?

**Review by fleupold — APPROVED** [review]

File: src/infra/dex/paraswap/mod.rs (hunk 1 above)
Comment (MartinquaXD) [reply src/infra/dex/paraswap/mod.rs:51]: Even if it doesn't it's more common to have UPPER-KEBAP-CASE for headers anyway, no?

Comment (squadgazzz) [reply src/infra/dex/paraswap/mod.rs:51]: Looks like it does not, since the lowercase is used in other solvers with the same nginx config.

**Review by MartinquaXD — APPROVED** [review]

File: src/infra/dex/paraswap/mod.rs (hunk 1 above)
Comment (MartinquaXD) [reply src/infra/dex/paraswap/mod.rs:51]: The [standard](https://stackoverflow.com/a/5259004) says it doesn't matter.

**Pull request merged**